module/VuFind/tests/unit-tests/src/VuFindTest/DeveloperSettings/DeveloperSettingsServiceTest.php

```diff
@@ -0,0 +1,380 @@
+<?php
+
+/**
+ * DeveloperSettingsService Test Class
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace VuFindTest\DeveloperSettings;
+
+use DateTime;
+use Generator;
+use VuFind\Db\Entity\ApiKey;
+use VuFind\Db\Entity\ApiKeyEntityInterface;
+use VuFind\Db\Entity\UserEntityInterface;
+use VuFind\Db\Service\ApiKeyServiceInterface;
+use VuFind\DeveloperSettings\DeveloperSettingsService;
+use VuFind\DeveloperSettings\DeveloperSettingsStatus;
+
+/**
+ * DeveloperSettingsService Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class DeveloperSettingsServiceTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Get a FavoritesService object.
+     *
+     * @param array                   $config        Configuration file presenting the API_Keys section
+     * @param ?ApiKeyServiceInterface $apiKeyService ApiKeyService mocked with methods
+     *
+     * @return DeveloperSettingsService
+     */
+    protected function getService(
+        array $config,
+        ?ApiKeyServiceInterface $apiKeyService = null
+    ): DeveloperSettingsService {
+        return new DeveloperSettingsService(
+            $apiKeyService ??= $this->createMock(ApiKeyServiceInterface::class),
```

Comment:
No need for an assignment here.

```suggestion
            $apiKeyService ?? $this->createMock(ApiKeyServiceInterface::class),
```